test/functional/terminal/tui_spec.lua

```diff
@@ -1919,6 +1919,39 @@ describe('TUI', function()
       eq({ 'https://example.com', '' }, exec_lua([[return _G.urls]]))
     end)
   end)
+
+  it('no heap-buffer-overflow when changing &columns', function()
+    -- Ensure a bg colour is set to trigger `print_spaces()` codepath in `clear_region()`.
+    child_session:request('nvim_command', 'highlight Normal ctermbg=red guibg=red')
+
+    feed_data(":call setline(1,['a'->repeat(&columns)])\n")
```

Comment:
Maybe `child_session:request('nvim_command', ...)` can also be used instead of `feed_data()` here?